src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/ColumnNames.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // ColumnNames(record:O): *[Value:s]
+    internal class ColumnNamesFunction_UO : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        private static readonly DType _returnType =
+            DType.CreateTable(new TypedName(DType.String, new DName(ColumnName_Value)));
+
+        public ColumnNamesFunction_UO()
+            : base("ColumnNames", TexlStrings.AboutColumnNames, FunctionCategories.Information, _returnType, 0, 1, 1, DType.UntypedObject)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.ColumnNamesArg1 };
+        }
+    }
+
+    // Column(record:O, property_name:s): O
+    internal class ColumnFunction_UO : BuiltinFunction
```

Comment:
It's a separate function. So probably needs to be in it's own file like other functions.

Reply:
We have examples of other functions bundled together with similar functions. DateTime.cs has lots of date/time related functions; UntypedObejcts.cs has ParseJSON and Index_UO; MinMax.cs has Min/Max; and some others. I don't have a problem splitting it out if you think this is important, though.

Reply:
In the past at least we covered function and all it's overloads in single file. It might have changed now.

Reply:
Split it into two different files